src/computability/regular_expressions.lean

```diff
@@ -0,0 +1,278 @@
+/-
+Copyright (c) 2020 Fox Thomson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Fox Thomson.
+-/
+import data.fintype.basic
+import data.finset.basic
+import tactic.rcases
+import tactic.omega
+
+/-!
+# Regular Expressions
+
+This file contains the formal definition for regular expressions and basic lemmas. Note these are
+regular expressions in terms of formal language theory. Note this is different to regex'x used in
+computer science such as the POSIX standard.
+-/
```

Comment:
Mind adding a link to https://en.wikipedia.org/wiki/Kleene_algebra in here somewhere?